### PR TITLE
Respect inlayhint padding

### PR DESCRIPTION
**Pull request by BoykoAlex**

### CODE REVIEW

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/inlayhint/LSPLineContentCodeMining.java [hunk 1]
```diff
@@ -65,10 +65,19 @@ public LSPLineContentCodeMining(InlayHint inlayHint, IDocument document,
...
 	@Override
 	public void setLabel(final @Nullable String label) {
-		if (label == null || label.isEmpty() || Character.isWhitespace(label.charAt(label.length() - 1)))
+		if (label == null) {
 			super.setLabel(label);
-		else
-			super.setLabel(label + " "); //$NON-NLS-1$
+		} else {
+			StringBuilder sb = new StringBuilder();
+			if (inlayHint.getPaddingLeft() != null && inlayHint.getPaddingLeft().booleanValue()) {
```
Comment (rubenporras) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/inlayhint/LSPLineContentCodeMining.java:72]:
```suggestion
			if (Boolean.TRUE.equals(inlayHint.getPaddingLeft()) {
```

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/inlayhint/LSPLineContentCodeMining.java [hunk 2]
```diff
@@ -65,10 +65,19 @@ public LSPLineContentCodeMining(InlayHint inlayHint, IDocument document,
...
+				sb.append(' ');
+			}
+			sb.append(label);
+			if (inlayHint.getPaddingRight() != null && inlayHint.getPaddingRight().booleanValue()) {
```
Comment (rubenporras) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/inlayhint/LSPLineContentCodeMining.java:76]:
```suggestion
			if (Boolean.TRUE.equals(inlayHint.getPaddingRight()) {
```

**Review by rubenporras — COMMENTED** [review]
What do you think of my suggestion?

For me, the code is shorter and easier to read.

**Comment by BoykoAlex** [comment]
@rubenporras I like your suggestions :-) I have folded them in :-)

**Comment by rubenporras** [comment]
Hi @BoykoAlex ,

the suggestions where not syntactically correct, I have missed a closing bracket in both, sorry for that.

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/inlayhint/LSPLineContentCodeMining.java [hunk 3]
```diff
@@ -65,10 +65,19 @@ public LSPLineContentCodeMining(InlayHint inlayHint, IDocument document,
...
 	@Override
 	public void setLabel(final @Nullable String label) {
-		if (label == null || label.isEmpty() || Character.isWhitespace(label.charAt(label.length() - 1)))
+		if (label == null) {
 			super.setLabel(label);
-		else
-			super.setLabel(label + " "); //$NON-NLS-1$
+		} else {
+			StringBuilder sb = new StringBuilder();
+			if (Boolean.TRUE.equals(inlayHint.getPaddingLeft()) {
```
Comment (rubenporras) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/inlayhint/LSPLineContentCodeMining.java:72]:
```suggestion
			if (Boolean.TRUE.equals(inlayHint.getPaddingLeft())) {
```

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/inlayhint/LSPLineContentCodeMining.java [hunk 4]
```diff
@@ -65,10 +65,19 @@ public LSPLineContentCodeMining(InlayHint inlayHint, IDocument document,
...
+				sb.append(' ');
+			}
+			sb.append(label);
+			if (Boolean.TRUE.equals(inlayHint.getPaddingRight()) {
```
Comment (rubenporras) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/inlayhint/LSPLineContentCodeMining.java:76]:
```suggestion
			if (Boolean.TRUE.equals(inlayHint.getPaddingRight())) {
```

**Comment by BoykoAlex** [comment]
@rubenporras no worries - so did I ;-)

**Pull request merged**